client/app/main/school/school.reports.controller.js

```diff
@@ -252,6 +252,102 @@ function SchoolReportsCtrl($scope, $timeout, Auth, uiGridGroupingConstants, Stud
     });
   };
 
+  $scope.outreachesGridOptions = {
+    rowHeight: 27,
+    enableSorting: true,
+    enableGridMenu: true,
+    enableFiltering: true,
+    treeRowHeaderAlwaysVisible: false,
+    exporterPdfDefaultStyle: {fontSize: 9},
+    exporterPdfTableHeaderStyle: {fontSize: 10, bold: true, color: 'grey'},
+    exporterPdfHeader: {
+      text: 'Chronically Absent Students',
+      style: 'headerStyle'
+    },
+    exporterPdfOrientation: 'landscape',
+    exporterPdfPageSize: 'LETTER',
+    exporterPdfMaxGridWidth: 500,
+    exporterPdfFooter: function(currentPage, pageCount) {
+      return {
+        text: currentPage.toString() + ' of ' + pageCount.toString(),
+        style: 'footerStyle'
+      };
+    },
+    exporterPdfCustomFormatter: function(docDefinition) {
+      docDefinition.styles.headerStyle =
+      {fontSize: 22, bold: true, color: '#265E6D'};
+      docDefinition.styles.footerStyle = {fontSize: 10, bold: true};
+      return docDefinition;
+    }
+  };
+
+  $scope.outreachesGridOptions.columnDefs = [{
+    name: 'student.studentId',
+    displayName: 'Student Id',
+    minWidth: 150,
+    cellTemplate: '<div class="ui-grid-cell-contents">' +
+                  '<a href="/student/{{row.entity.student._id}}">' +
+                  '{{row.entity.student.studentId}}</a>' +
+                  '</div>'
+    }, {
+      name: 'student.firstName',
+      displayName: 'First Name',
+      minWidth: 150
+    }, {
+      name: 'student.lastName',
+      displayName: 'Last Name',
+      minWidth: 150
+  }, {
+      name: 'total',
+      displayName: 'Total',
+      minWidth: 80
+  }, {
+      name: 'counts["Phone Call"]["count"] || 0',
+      displayName: 'Calls',
+      minWidth: 100
+  }, {
+      name: 'counts["Letter Sent"]["count"] || 0',
+      displayName: 'Letters',
+      minWidth: 100
+  }, {
+      name: 'counts["Home Visit"]["count"] || 0',
+      displayName: 'Visits',
+      minWidth: 100
+  }, {
+      name: 'counts["SST Referral"]["count"] || 0',
+      displayName: 'SST',
+      minWidth: 100
+  }, {
+      name: 'counts["Court Referral"]["count"] || 0',
+      displayName: 'Court',
+      minWidth: 100
+  }];
+
+  $scope.outreachesGridOptions.onRegisterApi = function(gridApi) {
+    $scope.outreachesGridApi = gridApi;
+    $scope.outreachesGridOptions.data = 
+      Student.outreachSummary();
+
+    $scope.outreachesGridOptions.data.$promise.then(function(data) {
+      // Convert counts array to object, generate total intervention property
+      $scope.outreachesGridOptions.data = _.map(data, 
```

Comment:
This feels kinda hacky. You are overwriting the value immediately after receiving it from the server. You can either, not set the value and use the promise directly to set it. Or (i prefer this), just use a _.forEach instead of map.

As a general rule, if you are using map you don't to mutate the value of the object that you are iterating since you are returning a new array.  

I would rewrite this _.map as:

```
      _.forEach(data, function(student) {
        student.total = _.sumBy(student.counts, 'count');
        student.counts = _.keyBy(student.counts, 'type');
      });
```

Or better yet not overwrite the "counts" and choose a different name. Notice that I have had to move the total calculation before the _.keyBy because overwriting counts reshapes the data returned from the server.